File: ruby/rotN.rb

def rand
num = Random.rand(13) + 1
puts "num = " + num.to_s
return num.to_i
end

def rotN(cha,num)
puts "character : "
#cha = gets
result = ""
cha.split(//).each_with_index do |s,i|
end

for i in 0..cha.length - 1 do
  if cha[i].ord > 'z'.ord - num then
    ans = ((cha[i].ord) - 26 + num).chr
  else
    ans = ((cha[i].ord) + num).chr
  end
  result += ans
end
puts result
return result
end

def main
  char = "kokoasdfrugb"
  rotN(char,rand)
end

main

File: python/rotN.py
import random

def rotN(N,cha):
    list = []
    for i in range(len(cha)):
        c = cha[i]
        num = ord(cha[i])
        if num > 0x7a - N:
            num = num - 26 + N
        else:
            num += N
        list.append(chr(num))
    result = "".join(list)
    print(result)
    return result

def main():
    print('今回はROT' + str(N) + 'です')

    print("文字列を入力してください")
    cha = input()

    rotN(N,cha)

if __name__ == "__main__":
    main()
